Makefile

```diff
@@ -94,22 +110,33 @@ data/list.eval: $(ALL_LSTMF)
 training: data/$(MODEL_NAME).traineddata
 
 ifdef START_MODEL
+$(TESSDATA_BEST)/$(START_MODEL).traineddata:
+	cd $(TESSDATA_BEST) && wget https://github.com/tesseract-ocr/tessdata$(TESSDATA_REPO)/raw/master/$(notdir $@)
 data/unicharset: $(ALL_BOXES)
 	mkdir -p data/$(START_MODEL)
-	combine_tessdata -u $(TESSDATA)/$(START_MODEL).traineddata  data/$(START_MODEL)/$(START_MODEL)
+	combine_tessdata -u $(TESSDATA_BEST)/$(START_MODEL).traineddata  data/$(START_MODEL)/$(START_MODEL)
 	unicharset_extractor --output_unicharset "$(GROUND_TRUTH_DIR)/my.unicharset" --norm_mode $(NORM_MODE) "$(ALL_BOXES)"
 	merge_unicharsets data/$(START_MODEL)/$(START_MODEL).lstm-unicharset $(GROUND_TRUTH_DIR)/my.unicharset  "$@"
 else
 data/unicharset: $(ALL_BOXES)
-	unicharset_extractor --output_unicharset "$@" --norm_mode 1 "$(ALL_BOXES)"
+	unicharset_extractor --output_unicharset "$@" --norm_mode $(NORM_MODE) "$(ALL_BOXES)"
 endif
 
 $(ALL_BOXES): $(sort $(patsubst %.tif,%.box,$(wildcard $(GROUND_TRUTH_DIR)/*.tif)))
 	find $(GROUND_TRUTH_DIR) -name '*.box' -exec cat {} \; > "$@"
 
-$(GROUND_TRUTH_DIR)/%.box: $(GROUND_TRUTH_DIR)/%.tif $(GROUND_TRUTH_DIR)/%.gt.txt
+ifeq ($(BOX_METHOD),WordStrBox)
+    $(GROUND_TRUTH_DIR)/%.box: $(GROUND_TRUTH_DIR)/%.tif $(GROUND_TRUTH_DIR)/%.gt.txt
+	tesseract "$(GROUND_TRUTH_DIR)/$*.tif" "$(GROUND_TRUTH_DIR)/$*" -l $(MODEL_NAME) --psm 6 -c tessedit_create_wordstrbox=1 
```

Comment:
This means using tesseract for generating the recognition atoms, right? How reliable is the segmentation?

Reply:
WordStrBox option creates a box file with the single line as a whole. I am hoping that tesseract will generate correct bounding boxes for single line emages.

examples:

WordStr 1 0 976 47 0 #ich denke. Aber was die ſelige Frau Geheimräthin

or

WordStr 0 0 922 48 0 #﻿Ἰάϰχε, Ἰάϰχε! Wie blitzen ihre großen Augen! Noch

or for a more relevant example using Devanagari script (Indic)

WordStr 0 0 1450 100 0 #तर्क एव सहायोऽभूत् तर्क एवास्ति बाधकः ॥

Reply:
I looked at the python script just now. It is possible to create the WordStrBox file as follows:

```
# create WordStr line boxes for Indic & RTL
for line in lines:
    if line.strip():
        print(u"%s %d %d %d %d 0 #%s" % ("WordStr", 0, 0, width, height, line))
        print(u"%s %d %d %d %d 0" % ("\t", width, height, width+1, height+1))
```

In fact even the second line with tab may not be required.